dd-sdk-android/src/main/kotlin/com/datadog/android/v2/core/DatadogCore.kt

```diff
@@ -92,31 +104,38 @@ internal class DatadogCore(
         storageConfiguration: FeatureStorageConfiguration,
         uploadConfiguration: FeatureUploadConfiguration
     ) {
-        // TODO RUMM-0000 read configuration and create storage and uploader instead of taking
-        //  from already initialized SdkFeature
-        val (storage, uploader) = when (featureName) {
-            RumFeature.RUM_FEATURE_NAME ->
-                rumFeature?.persistenceStrategy?.getStorage() to rumFeature?.uploader
-            LogsFeature.LOGS_FEATURE_NAME ->
-                logsFeature?.persistenceStrategy?.getStorage() to logsFeature?.uploader
-            TracingFeature.TRACING_FEATURE_NAME ->
-                tracingFeature?.persistenceStrategy?.getStorage() to tracingFeature?.uploader
-            CrashReportsFeature.CRASH_FEATURE_NAME ->
-                crashReportsFeature?.persistenceStrategy?.getStorage() to
-                    crashReportsFeature?.uploader
-            WebViewRumFeature.WEB_RUM_FEATURE_NAME ->
-                webViewRumFeature?.persistenceStrategy?.getStorage() to webViewRumFeature?.uploader
-            WebViewLogsFeature.WEB_LOGS_FEATURE_NAME ->
-                webViewLogsFeature?.persistenceStrategy?.getStorage() to
-                    webViewLogsFeature?.uploader
-            else -> {
-                null to null
+        val storage = createStorage(featureName, storageConfiguration)
+        val uploader = DataOkHttpUploader(
+            requestFactory = uploadConfiguration.requestFactory,
+            internalLogger = sdkLogger,
+            callFactory = coreFeature.okHttpClient,
+            sdkVersion = coreFeature.sdkVersion,
+            androidInfoProvider = coreFeature.androidInfoProvider
+        )
+        // TODO RUMM-0000 those still need to be registered explicitly,
+        //  because of cross-functionality. We will get rid of them eventually and use getFeature
+        when (featureName) {
```

Comment:
are we sure they are not stepping each other foot ? Meaning that now you have 2 features for same feature (e.g. for RUM you have `rumFeature` and `DatadogFeature`)

Reply:
currently no, because `DatadogFeature` is just a holder for now, it doesn't have a lifecycle. In the future I'm thinking about the possibility of getting rid of `DatadogFeature` and put things into `SdkFeature` instead.